publication-event-dtopublicationdoi-producer/src/main/java/no/unit/nva/doi/event/producer/DynamoDbFanoutPublicationDtoProducer.java

```diff
@@ -0,0 +1,65 @@
+package no.unit.nva.doi.event.producer;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.events.DynamodbEvent;
+import java.util.Optional;
+import no.unit.nva.events.handlers.EventHandler;
+import no.unit.nva.events.models.AwsEventBridgeEvent;
+import no.unit.nva.publication.doi.PublicationMapper;
+import no.unit.nva.publication.doi.dto.PublicationHolder;
+import no.unit.nva.publication.doi.dto.PublicationMapping;
+import nva.commons.utils.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Consumes DynamodbEvent's that's been published on EventBridge, and produces new PublicationCollection DTO with type
+ * `doi.publication`.
+ */
+public class DynamoDbFanoutPublicationDtoProducer
+    extends EventHandler<DynamodbEvent.DynamodbStreamRecord, PublicationHolder> {
+
+    public static final String TYPE_DTO_DOI_PUBLICATION = "doi.publication";
+    public static final PublicationHolder NO_OUTPUT_NO_EVENT = null;
+    private static final Logger logger = LoggerFactory.getLogger(DynamoDbFanoutPublicationDtoProducer.class);
+    private final PublicationMapper publicationMapper;
+
+    @JacocoGenerated
+    public DynamoDbFanoutPublicationDtoProducer() {
+        this(AppConfig.getNamespace());
+    }
+
+    public DynamoDbFanoutPublicationDtoProducer(String namespace) {
+        super(DynamodbEvent.class);
+        this.publicationMapper = new PublicationMapper(namespace);
+    }
+
+    @Override
+    protected PublicationHolder processInput(DynamodbEvent.DynamodbStreamRecord input,
+                                             AwsEventBridgeEvent<DynamodbEvent.DynamodbStreamRecord> event,
+                                             Context context) {
+        return fromDynamodbStreamRecords(input);
+    }
+
+    private PublicationHolder fromDynamodbStreamRecords(DynamodbEvent.DynamodbStreamRecord record) {
+        var dto = Optional.ofNullable(record)
+            .map(publicationMapper::fromDynamodbStreamRecord)
+            .filter(this::isEffectiveChange)
+            .map(publicationMapping -> publicationMapping.getNewPublication().orElseThrow());
+
+        logger.info("{} Publication DTO from DynamodbStreamRecord", dto.isPresent() ? "Created" : "Skipped creating");
+        return dto
```

Comment:
Less important than previous, but maybe extract this one to method too

Reply:
Fixed.